pkg/ccl/testccl/sqlccl/run_control_test.go

```diff
@@ -205,6 +205,29 @@ func testCancelSession(t *testing.T, hasActiveSession bool) {
 					_, err = conn1.ExecContext(ctx, "SELECT pg_sleep(1000000)")
 					errChan <- err
 				}()
+				// Block until the query goroutine was spun up and began
+				// executing the query - this is needed to avoid a race between
+				// canceling the session before vs after 'pg_sleep' query begins
+				// (the former would result in an unexpected error message).
+				toSleep := time.Millisecond
```

Comment:
nit: this might be simplified a little with `testutils.SucceedsSoon`. here's an existing example: https://github.com/cockroachdb/cockroach/blob/3cd24e8377c219b1dd05ef612e0366bd5c3c3dad/pkg/sql/show_test.go#L695-L715